client/app/views/steps/welcome.coffee

```diff
@@ -11,9 +25,29 @@ module.exports = class WelcomeView extends StepView
     events:
         'click @ui.next': 'onSubmit'
 
+
     onSubmit: (event) ->
         event.preventDefault()
-        @model.submit()
+        @model.submit() unless @isDisabled()
+        @setDisabled()
+
+
+    setDisabled: ($el = @ui.next) ->
+        if $el[0].tagName.toLowerCase() in FORMS_DIS_ELEMENTS
+            $el
+                .prop 'disabled', true
+                .attr 'aria-busy', true
+        else
+            $el.attr
+                'aria-disabled': true
+                'aria-busy':     true
+
+
+    isDisabled: ($el = @ui.next) ->
+        if $el[0].tagName.toLowerCase() in FORMS_DIS_ELEMENTS
```

Comment:
I understand your argument but I can't figure out if it's really releant or not: using a script like this one, which allaow to submit again, is clearly an volunteer, malintentioned action. I'm not sure if it a good reason to store Ui states in JS instead of using the markup. But your opinion seems reasonnable… I need to think about it a little more…